cipher/affine.c

```diff
@@ -0,0 +1,207 @@
+/**
+ * @file
+ * @brief An [affine cipher](https://en.wikipedia.org/wiki/Affine_cipher) is a
+ * letter substitution cipher that uses a linear transformation to substitute
+ * letters in a message.
+ * @details Given an alphabet of length M with characters with numeric values
+ * 0-(M-1), an arbitrary character x can be transformed with the expression (ax
+ * + b) % M into our ciphertext character. The only caveat is that a must be
+ * relatively prime with M in order for this transformation to be invertible,
+ * i.e., gcd(a, M) = 1.
+ * @author [Daniel Murrow](https://github.com/dsmurrow)
+ */
+
+#include <assert.h>  /// for assertions
+#include <stdio.h>   /// for IO
+#include <stdlib.h>  /// for div function and div_t struct as well as malloc and free
+#include <string.h>  /// for strlen, strcpy, and strcmp
+
+/**
+ * @brief number of characters in our alphabet (printable ASCII characters)
+ */
+#define ALPHABET_SIZE 95
+
+/**
+ * @brief used to convert a printable byte (32 to 126) to an element of the
+ * group Z_95 (0 to 94)
+ */
+#define Z95_CONVERSION_CONSTANT 32
+
+/**
+ * @brief a structure representing an affine cipher key
+ */
+typedef struct
+{
+    int a;  ///< what the character is being multiplied by
+    int b;  ///< what is being added after the multiplication with a
+} affine_key_t;
+
+/**
+ * @brief finds the value x such that (a * x) % m = 1
+ *
+ * @param a number we are finding the inverse for
+ * @param m the modulus the inversion is based on
+ *
+ * @returns the modular multiplicative inverse of a mod m
```

Comment:
I believe these two are variables/parameters.
```suggestion
 * @returns the modular multiplicative inverse of `a` mod `m`
```